TestingLowerBounds/KullbackLeibler.lean

```diff
@@ -171,14 +172,33 @@ lemma kl_eq_zero_iff [SigmaFinite μ] [SigmaFinite ν] : kl μ ν = 0 ↔ μ = 
     by_cases h_int : Integrable (llr μ ν) μ
     swap; · rw [kl_of_not_integrable h_int] at h; simp_all only [EReal.top_ne_zero]
     sorry -- TODO : decide what proof strategy to use here, maybe we could use the fact that jensen's inequality is an equality iff the function is constant a.e., but I don't know wether this is in mathlib
-  · rw [h]
-    exact kl_self ν
+  · exact h ▸ kl_self ν
 
 end kl_nonneg
 
 section Conditional
 
-variable {β : Type*} {mβ : MeasurableSpace β} {κ η : kernel α β} {μ : Measure α}
+variable {β : Type*} {mβ : MeasurableSpace β} {κ η : kernel α β}
+
+/--Equivalence between two possible versions of the first condition for the finiteness of the
+conditional KL divergence, the second version is the preferred one.-/
+lemma kl_ae_ne_top_iff : (∀ᵐ a ∂μ, kl (κ a) (η a) ≠ ⊤) ↔
+    (∀ᵐ a ∂μ, κ a ≪ η a) ∧ (∀ᵐ a ∂μ, Integrable (llr (κ a) (η a)) (κ a)) := by
+  constructor <;> intro h
+  · constructor <;> filter_upwards [h] with a ha <;> have := kl_eq_top_iff.mpr.mt ha <;> tauto
+  · filter_upwards [h.1, h.2] with a ha1 ha2
+    apply kl_eq_top_iff.mp.mt
+    tauto
```

Comment:
Untested, but something along that line should work, no?
```suggestion
  simp_rw [ne_eq, kl_eq_top_iff, Filter.eventually_and]
```

Reply:
Given that this lemma is really just putting an almost everywhere around `kl_eq_top_iff`, perhaps it's not needed?

Reply:
In the cases where we just rewrite the lemma it could indeed be substituted by this one line proof, but I think it would be harder to use it without the wrapped version when we need it inside the argument of some other lemma, like in the proof of 'condKL_of_ae_ac_of_ae_integrable_of_integrable':
'condKL_of_ae_ne_top_of_integrable (kl_ae_ne_top_iff.mpr ⟨h_ac, h_ae_int⟩) h_int'